source/bootloader_common.c

```diff
@@ -29,15 +29,70 @@ const char hexTable[16] = {'0', '1', '2', '3', '4', '5', '6', '7',
                            '8', '9', 'A', 'B', 'C', 'D', 'E', 'F'
                           };
 
+
+#if DEVICE_SERIAL
+
+#include "hal/serial_api.h"
+
+static serial_t stdio_uart = { 0 };
+
+/* module variable for keeping track of initialization */
+static bool not_initialized = true;
+
+/**
+ * @brief Initialization serial port if needed.
+ *.
+ */
+static void init_serial()
+{
+    if (not_initialized)
+    {
+        not_initialized = false;
+
+        serial_init(&stdio_uart, STDIO_UART_TX, STDIO_UART_RX);
+#if MBED_CONF_PLATFORM_STDIO_BAUD_RATE
+        serial_baud(&stdio_uart, MBED_CONF_PLATFORM_STDIO_BAUD_RATE);
+#endif
+    }
+
+}
+
+/**
+ * @brief Function that directly outputs to serial port in blocking mode.
+ *
+ * @param string outputed to serial port.
+ */
+void boot_debug(const char *s)
+{
+	init_serial();
+
+	for(int i = 0; i < strlen(s); i++)
```

Comment:
Instead of using `strlen()` you could write this into a form of while loop:
```
while(*s) {
    serial_putc(, *s);
    s++;
}
```

one more stdlib function dropped..